MekHQ/src/mekhq/campaign/stratcon/StratconContractInitializer.java

```diff
@@ -105,17 +104,28 @@ public static void initializeCampaignState(AtBContract contract, Campaign campai
                         break;
                     case AlliedFacilityControl:
                         initializeTrackFacilities(campaignState.getTrack(x), numObjects, ForceAlignment.Allied, true);
-                        campaignState.incrementStrategicObjectiveCompletedCount(numObjects);
                         break;
                     case HostileFacilityControl:
                     case FacilityDestruction:
                         initializeTrackFacilities(campaignState.getTrack(x), numObjects, ForceAlignment.Opposing, true);
                         break;
                     case AnyScenarioVictory:
-                        if (objectiveParams.objectiveScenarioModifiers != null) {
-                            campaignState.getGlobalScenarioModifiers().addAll(objectiveParams.objectiveScenarioModifiers);
-                        }
-                    break;
+                        // set up a "win X scenarios" objective
+                        StratconStrategicObjective sso = new StratconStrategicObjective();
+                        sso.setDesiredObjectiveCount(numObjects);
+                        sso.setObjectiveType(StrategicObjectiveType.AnyScenarioVictory);
+                        campaignState.getTrack(x).addStrategicObjective(sso);
+                        break;
+                }
+            }
```

Comment:
Getting a warning for a missing case from CI. Does RequiredScenarioVictory need to be handled here?

Reply:
No - currently, none of the contract definitions have it. I was briefly considering having the "required" scenarios from liaison+ command be a separate objective, but shelved that idea.

Reply:
Solid idea for the future, especially when considering the CO descriptions